flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/InterruptSensitiveRestoreTest.java

```diff
@@ -280,4 +368,50 @@ public void restoreState(Serializable state) throws Exception {
 			fail("should never be called");
 		}
 	}
+
+	private static class TestSourceOperator implements SourceFunction<Object>, CheckpointedFunction {
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public void run(SourceContext<Object> ctx) throws Exception {
+			fail("should never be called");
+		}
+
+		@Override
+		public void cancel() {}
+
+
+		@Override
+		public void snapshotState(FunctionSnapshotContext context) throws Exception {
+			fail("should never be called");
+		}
+
+		@Override
+		public void initializeState(FunctionInitializationContext context) throws Exception {
+			((StateInitializationContext)context).getRawOperatorStateInputs().iterator().next().getStream().read();
+		}
+	}
+
+	private static class TestSourceKeyed implements SourceFunction<Object>, CheckpointedFunction {
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public void run(SourceContext<Object> ctx) throws Exception {
+			fail("should never be called");
+		}
+
+		@Override
+		public void cancel() {}
+
+
+		@Override
+		public void snapshotState(FunctionSnapshotContext context) throws Exception {
+			fail("should never be called");
+		}
+
+		@Override
+		public void initializeState(FunctionInitializationContext context) throws Exception {
+			((StateInitializationContext)context).getRawKeyedStateInputs().iterator().next().getStream().read();
```

Comment:
Even with this line commented out the tests still succeed. Probably because the `AbstractStreamOperator` already tries reading from the keyed streams. What do you think?